### PR TITLE
feat(server): use pg_dumpall version that matches the database version

### PR DESCRIPTION
**Pull request by zackpollard**

### CODE REVIEW

**Review by jrasm91 — APPROVED** [review]

File: server/src/services/backup.service.ts [hunk 1]
```diff
@@ -101,14 +101,23 @@ export class BackupService extends BaseService {
       `immich-db-backup-${Date.now()}.sql.gz.tmp`,
     );
 
+    const databaseVersion = await this.databaseRepository.getPostgresVersion();
+    const databaseMajorVersion = Number.parseInt(databaseVersion.split('.')[0], 10);
```
Comment (danieldietzler) [inline server/src/services/backup.service.ts:105]: It's semver, right? Should we use the semver package instead then?

**Review by danieldietzler — APPROVED** [review]

File: server/src/services/backup.service.ts (hunk 1 above)
Comment (zackpollard) [reply server/src/services/backup.service.ts]: Moved to checking a range with semver

File: docker/docker-compose.dev.yml
Comment (bo0tzz) [inline docker/docker-compose.dev.yml:1]: Are these changes intentional?

Comment (zackpollard) [reply docker/docker-compose.dev.yml:1]: Just trying to fix these damn semver changes :laughing: Accidentally committed those, will cleanup in a minute
